stl/inc/chrono

```diff
@@ -3472,66 +3472,49 @@ namespace chrono {
             const _LeapSecondRep _Original_leap{_Leap};
             if (_Leap == _LeapSecondRep::_File_time) {
                 const int _Year{*_Century + *_Two_dig_year};
-                _Leap = _Year <= 2016 ? _LeapSecondRep::_Negative : _LeapSecondRep::_All;
+                _Leap = _Year <= 2016 ? _LeapSecondRep::_None : _LeapSecondRep::_All;
             }
 
             if (_Second > (_Leap == _LeapSecondRep::_All ? 60 : 59)) {
                 return false;
             }
 
-            // A distinction has to be made here between clocks that tick monotonically, even around a positive leap
-            // second (everything except system_clock) and ones that have a special representation for leap seconds
-            // (utc_clock and, for negative leap seconds, system_clock). gps_clock and tai_clock, in particular, always
-            // monotonically count 86,400 seconds/day. The correct tick count, therefore, can be determined without
+            // A distinction has to be made here between clocks that always count 86,400 seconds/day (system_clock,
+            // tai_clock, and gps_clock) and ones that count leap seconds (utc_clock and, since 2018-07-01, file_clock).

```

Comment:
I tested what date/time `w32tm` accepts for force-adding leap seconds. The result suggests that 2018-06-01 is more likely the date when Windows started counting leap seconds. I'll update the code and comment accordingly.

```
C:\WINDOWS\system32>w32tm /leapseconds /add:+2018-05-31T23:59:59 /force /testmode
分析给定的日期时间字符串时出错: +2018-05-31T23:59:59
Please make sure it is in the format [+-]YYYY-MM-ddThh:mm and represents a valid datetime.
Examples: +2018-12-31T23:59, -2019-06-30T23:59, +2019-12-31T23:59:59
Failed to add leap second either due to input parsing failure or because there are identical or newer leap second(s) in the settings.

C:\WINDOWS\system32>w32tm /leapseconds /add:+2018-06-01T00:59:59 /force /testmode
成功地执行了命令。

C:\WINDOWS\system32>w32tm /leapseconds /getstatus /verbose
[闰秒]
Enabled: 1 (Local)
Number of Leap Seconds (after June 2018): 1 (Local)
Leap Seconds List (Local):
+2018-06-01T00:59:59
```

Reply:
Having checked the Windows source code, I believe that @statementreply is exactly correct. (I can't quote the code here, and to be cautious I won't even mention the source path, but I found an explicit test in the code and I believe it's directly responsible for this observed behavior.)